library/src/G4CMPPhononBoundaryProcess.cc

```diff
@@ -439,15 +453,16 @@ GetReflectedVector(const G4ThreeVector& waveVector,
 // Generate diffuse reflection according to 1/cos distribution
 
 G4ThreeVector G4CMPPhononBoundaryProcess::
-GetLambertianVector(const G4ThreeVector& surfNorm, G4int mode) const {
+GetLambertianVector(const G4ThreeVector& surfNorm, G4int mode,
+                    const G4ThreeVector& surfPoint) const {
   G4ThreeVector reflectedKDir;
   const G4int maxTries = 1000;
   G4int nTries = 0;
   do {
     reflectedKDir = G4CMP::LambertReflection(surfNorm);
   } while (nTries++ < maxTries &&
 	   !G4CMP::PhononVelocityIsInward(theLattice, mode,
-					  reflectedKDir, surfNorm));
+					  reflectedKDir, surfNorm, surfPoint));
 
   return reflectedKDir;
 }
```

Comment:
Could we move this whole function out of the boundary process?  Maybe to `G4CMPUtils`, with `PhononVelocityIsInward()`?   If we do that, then PhononElectrode could just call it, rather than having to have a copy of the implementation.